src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7992,6 +8034,25 @@ internal static bool IsReparsePoint(FileSystemInfo fileInfo)
                 : Platform.NonWindowsIsSymLink(fileInfo);
         }
 
+        internal static bool IsNameSurrogateReparsePoint(string filePath)
+        {
+            if (Platform.IsWindows)
+            {
+                var data = new WIN32_FIND_DATA();
+                using (SafeFileHandle handle = FindFirstFileEx(filePath, FINDEX_INFO_LEVELS.FindExInfoBasic, ref data, FINDEX_SEARCH_OPS.FindExSearchNameMatch, IntPtr.Zero, 0))
+                {
+                    // Name surrogates are reparse points that point to other named entities local to the filesystem (like symlinks)
+                    // In the case of OneDrive, they are not surrogates and would be safe to recurse into.
+                    if (!handle.IsInvalid && (data.dwReserved0 & 0x20000000) == 0)
```

Comment:
Is it ok for compliance?
Related https://github.com/dotnet/corefx/issues/26310

Reply:
This code is equivalent to this macro in a published header https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/ntifs/nf-ntifs-isreparsetagnamesurrogate

Reply:
I think we could add this to the comment because dwReserved0 confuses.